backend/ops_api/ops/document/api.py

```diff
@@ -0,0 +1,59 @@
+from flask import Blueprint, Response, current_app, request
+
+from ops_api.ops.document.document_gateway import DocumentGateway
+from ops_api.ops.document.exceptions import ValidationError
+from ops_api.ops.document.schema import DocumentRequestSchema, DocumentResponseSchema, GetDocumentRequestSchema
+from ops_api.ops.document.service import DocumentService
+from ops_api.ops.utils.response import make_response_with_headers
+
+# Create Blueprint object
+documents_bp = Blueprint("documents", __name__)
+
+
+@documents_bp.route("/documents/", methods=["GET"])
+def get_documents_by_agreement_id():
+
+    # Validate request data
+    request_schema = GetDocumentRequestSchema()
+    try:
+        request_data = request_schema.dump(request_schema.load(request.args))
+        request_schema.validate(request_data)
+    except ValidationError as e:
+        return make_response_with_headers({"message": "Validation Error", "errors": e}, 400)
+
+    # Get agreement_id from request
+    agreement_id = request.args.get("agreement_id")
+
+    if not agreement_id:
```

Comment:
shouldn't need this because `agreement_id` is marked `required=True` in the schema.  If `agreement_id` is missing in the request a `ValidationError` will be thrown from line 19